lib/ruby_smb/server/cli.rb

```diff
@@ -0,0 +1,121 @@
+        defaults = DEFAULT_OPTIONS.merge(defaults)
+        options = defaults.clone
+        OptionParser.new do |parser|
+          parser.on("--share SHARE", "The share name (default: #{defaults[:share_name]})") do |share|
+            options[:share_name] = share
+          end
+
+          parser.on("--[no-]anonymous", "Allow anonymous access (default: #{defaults[:allow_anonymous]})") do |allow_anonymous|
+            options[:allow_anonymous] = allow_anonymous
+          end
+
+          parser.on("--[no-]guests", "Allow guest accounts (default: #{defaults[:allow_guests]})") do |allow_guests|
+            options[:allow_guests] = allow_guests
+          end
+
+          parser.on("--[no-]smbv1", "Enable or disable SMBv1 (default: #{defaults[:smbv1] ? 'Enabled' : 'Disabled'})") do |smbv1|
+            options[:smbv1] = smbv1
+          end
+
+          parser.on("--[no-]smbv2", "Enable or disable SMBv2 (default: #{defaults[:smbv2] ? 'Enabled' : 'Disabled'})") do |smbv2|
+            options[:smbv2] = smbv2
+          end
+
+          parser.on("--[no-]smbv3", "Enable or disable SMBv3 (default: #{defaults[:smbv3] ? 'Enabled' : 'Disabled'})") do |smbv3|
+            options[:smbv3] = smbv3
+          end
+
+          parser.on("--username USERNAME", "The account's username (default: #{defaults[:username]})") do |username|
+            if username.include?('\\')
+              options[:domain], options[:username] = username.split('\\', 2)
+            else
+              options[:username] = username
+            end
+          end
+
+          parser.on("--password PASSWORD", "The account's password (default: #{defaults[:password]})") do |password|
+            options[:password] = password
+          end
+
+          block.call(options, parser) if block_given?
+        end.parse!
+
+        options
+      end
+
+      # Build a server instance from the specified options. The NTLM provider will be used for authentication.
+      #
+      # @param [Hash] options the options to use while building the server. See the return value of {parse} for a
+      #   comprehensive list of keys.
+      def self.build(options)
+        ntlm_provider = RubySMB::Gss::Provider::NTLM.new(
+          allow_anonymous: options[:allow_anonymous],
+          allow_guests: options[:allow_guests]
+        )
+        ntlm_provider.put_account(options[:username], options[:password], domain: options[:domain])  # password can also be an NTLM hash
+
+        server = RubySMB::Server.new(
+          gss_provider: ntlm_provider,
+          logger: :stdout
+        )
+        server.dialects.select! { |dialect| RubySMB::Dialect[dialect].family != RubySMB::Dialect::FAMILY_SMB1 } unless options[:smbv1]
+        server.dialects.select! { |dialect| RubySMB::Dialect[dialect].family != RubySMB::Dialect::FAMILY_SMB2 } unless options[:smbv2]
+        server.dialects.select! { |dialect| RubySMB::Dialect[dialect].family != RubySMB::Dialect::FAMILY_SMB3 } unless options[:smbv3]
```

Comment:
Not a blocker, this might read better as `filter!`

```suggestion
        server.dialects.filter! { |dialect| RubySMB::Dialect[dialect].family != RubySMB::Dialect::FAMILY_SMB3 } unless options[:smbv3]
```